controller/linodemachine_controller.go

```diff
@@ -253,108 +250,210 @@ func (r *LinodeMachineReconciler) reconcileCreate(
 			logger.Error(err, "Failed to boot instance")
 
-			return nil, err
+			return ctrl.Result{}, err
 		}
+		logger.Info("Booting instance")
+
 	default:
-		err = errors.New("multiple instances")
+		err := errors.New("multiple instances")
 		logger.Error(err, "multiple instances found", "tags", tags)
 
-		return nil, err
+		return ctrl.Result{}, err
+	}
+
+	machineScope.LinodeMachine.Status.Ready = true
+	machineScope.LinodeMachine.Spec.InstanceID = &linodeInstance.ID
+	machineScope.LinodeMachine.Spec.ProviderID = util.Pointer(fmt.Sprintf("linode://%d", linodeInstance.ID))
+	machineScope.LinodeMachine.Status.Addresses = buildInstanceAddrs(linodeInstance)
+
+	return ctrl.Result{}, nil
+}
+
+func (r *LinodeMachineReconciler) reconcileCreateControlNode(
+	ctx context.Context,
+	logger logr.Logger,
+	machineScope *scope.MachineScope,
+	linodeInstances []linodego.Instance,
+	tags []string,
+) (ctrl.Result, error) {
+	var linodeInstance *linodego.Instance
+	var instanceConfig *linodego.InstanceConfig
+
+	switch len(linodeInstances) {
+	case 1:
+		logger.Info("Linode instance already exists")
+
+		linodeInstance = &linodeInstances[0]
+
+		configs, err := machineScope.LinodeClient.ListInstanceConfigs(ctx, linodeInstance.ID, &linodego.ListOptions{})
+		if err != nil || len(configs) == 0 {
+			logger.Error(err, "Failed to list instance configs")
+
+			return ctrl.Result{}, err
+		}
+		instanceConfig = &configs[0]
+
+	case 0:
+		// get the bootstrap data for the Linode instance and set it for create config
+		createOpts, err := r.newCreateConfig(ctx, machineScope, tags, logger)
+		if err != nil {
+			logger.Error(err, "Failed to create Linode machine InstanceCreateOptions")
+
+			return ctrl.Result{}, err
+		}
+
+		// Omit image and interfaces when creating the instance to configure disks and config profile manually
+		image := createOpts.Image
+		createOpts.Image = ""
+		interfaces := createOpts.Interfaces
+		createOpts.Interfaces = nil
+
+		linodeInstance, err = machineScope.LinodeClient.CreateInstance(ctx, *createOpts)
+		if err != nil || linodeInstance == nil {
+			logger.Error(err, "Failed to create Linode machine instance")
+
+			return ctrl.Result{}, err
+		}
+
+		createOpts.Image = image
+		createOpts.Interfaces = interfaces
+		instanceConfig, err = r.configureControlPlane(ctx, logger, machineScope, linodeInstance.ID, *createOpts)
+		if err != nil {
+			logger.Error(err, "Failed to configure instance profile")
+
+			return ctrl.Result{}, err
+		}
+
+	default:
+		err := errors.New("multiple instances")
+		logger.Error(err, "multiple instances found", "tags", tags)
+
+		return ctrl.Result{}, err
+	}
+
+	ok, err := r.checkControlPlaneDiskStatuses(ctx, machineScope, linodeInstance.ID, instanceConfig)
+	if err != nil {
+		logger.Error(err, "Failed to check instance disks statuses")
+
+		return ctrl.Result{RequeueAfter: reconciler.DefaultMachineControllerWaitForControlPlaneDisksDelay}, err
```

Comment:
I'm not sure if we necessarily want to requeue if there was an error on the API side unless it was a transient error. We should probably just do `return ctrl.Result{}, err` like in the other error cases for now.